File: ruby/spec/doubles/test_order_repository.rb
# frozen_string_literal: true

class TestOrderRepository
  def initialize
    @orders = []
    @inserted_order = nil
  end

  def saved_order
    @inserted_order
  end

  def save(order)
    @inserted_order = order
  end

  def get_by_id(order_id)
    @orders.first { |o| o.id == order_id }
  end

  def add_order(order)
    @orders << order
  end
end

File: scala/src/test/scala/doubles/TestOrderRepository.scala
package doubles

import ordershipping.domain.Order
import ordershipping.repository.OrderRepository

import scala.collection.mutable

class TestOrderRepository extends OrderRepository {
  private var insertedOrder: Order = _
  private val orders: mutable.MutableList[Order] = mutable.MutableList.empty

  override def save(order: Order): Unit =
    insertedOrder = order

  override def getById(orderId: Int): Option[Order] =
    orders.find(o => o.id == orderId)

  def addOrder(order: Order): Unit = orders += order
  def savedOrder(): Order = insertedOrder
}
